Essentials/src/net/ess3/api/events/MuteStatusChangeEvent.java

```diff
@@ -4,7 +4,26 @@
 
 
 public class MuteStatusChangeEvent extends StatusChangeEvent {
-    public MuteStatusChangeEvent(IUser affected, IUser controller, boolean value) {
+    private Long timestamp;
+    private String reason;
+
+    public MuteStatusChangeEvent(IUser affected, IUser controller, boolean value, long timestamp, String reason) {
         super(affected, controller, value);
+        this.timestamp = timestamp;
+        this.reason = reason == null ? null : (reason.isEmpty() ? null : reason);
+    }
+
+    /**
+     * @return If the mute is temporary, returns the timestamp; if permanent, returns 0, and if unknown, returns -1.
+     */
+    public Long getTimestamp() {
```

Comment:
I'm not sure whether we should return 0 or null if the mute is permanent. `null` better describes that there isn't a timestamp for the mute to be lifted on. @Ichbinjoe any thoughts?

(I'm aware I've changed my mind on this since my last look over this PR - apologies about the changes.)

Reply:
Sorry for delay (disappeared for some time).

I think the optimal solution is to potentially wrap this Long in an Optional<Long>. It beats readers over the head with the intent of the field.

Would opt against 0 - easy to mess up. Null would at least cause things to blow up in the case where there is no timestamp.